analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs

```diff
@@ -221,14 +225,29 @@ public void SetParameterValues_WithWrongPropertyType_StringInsteadOfInt_DoesNotP
         public void SetParameterValues_WithWrongPropertyType_StringInsteadOfBoolean_DoesNotPopulateProperties()
         {
             // Arrange
-            var options = AnalysisScaffolding.CreateOptions("ResourceTests\\StringInsteadOfBoolean\\SonarLint.xml");
+            var compilation = CreateCompilationWithOption("ResourceTests\\StringInsteadOfBoolean\\SonarLint.xml");
             var analyzer = new CheckFileLicense(); // Cannot use mock because we use reflection to find properties.
 
             // Act
-            ParameterLoader.SetParameterValues(analyzer, options);
+            ParameterLoader.SetParameterValues(analyzer, compilation.SonarLintFile());
 
             // Assert
             analyzer.IsRegularExpression.Should().BeFalse(); // Default value
         }
+
+        private static SonarCompilationReportingContext CreateCompilationWithOption(string filePath, SourceText text = null)
+        {
+            var options = text is null
+                ? AnalysisScaffolding.CreateOptions(filePath)
+                : AnalysisScaffolding.CreateOptions(filePath, text);
+            var compilation = SolutionBuilder.Create().AddProject(AnalyzerLanguage.CSharp).GetCompilation();
+            return CreateCompilation(compilation, options);
+        }
+
+        private static SonarCompilationReportingContext CreateCompilation(Compilation compilation, AnalyzerOptions options)
```

Comment:
This method is only used in the one above, is it really needed to be a separate method? I would say it can be merged together with `CreateCompilationWithOption`.